scipy/integrate/_quadrature.py

```diff
@@ -1029,20 +1031,22 @@ def cumulative_simpson(y, *, x=None, dx=1.0, axis=-1, initial=None):
     y = _ensure_float_array(y)
 
     # validate `axis` and standardize to work along the last axis
+    original_y = y.copy()
     original_shape = y.shape
     try:
         y = np.swapaxes(y, axis, -1)
     except IndexError as e:
         message = f"`axis={axis}` is not valid for `y` with `y.ndim={y.ndim}`."
         raise ValueError(message) from e
+    
+    if y.shape[-1] == 0:
+        raise ValueError("At least 1 point is required along axis.")
 
-    if y.shape[-1] < 3:
-        raise ValueError(
-            "At least 3 points are required along the axis of integration "
-            "to use the composite Simpson's method."
-        )
+    elif 0 < y.shape[-1] < 3:
+        res = cumulative_trapezoid(original_y, x, dx=dx, axis=axis, initial=None)
+        res = np.swapaxes(res, axis, -1)
```

Comment:
Why is `original_y` needed here?

Reply:
At this point, we have performed np.swapaxes for y, but not for x, dx, or initial.
The axis of integration has to be consistent across all inputs, so we need original_y to pass along to cumulative_trapezoid.